Framework/Algorithms/src/RadiusSum.cpp

```diff
@@ -148,12 +149,12 @@ std::vector<double> RadiusSum::processNumericImageRadiusSum() {
   // or the center of the bin, if it is a histogram.
 
   g_log.debug() << "Define the X positions of the pixels\n";
-  auto &refX = inputWS->getSpectrum(0).readX();
-  auto &refY = inputWS->getSpectrum(0).readY();
+  auto &refX = inputWS->getSpectrum(0).x();
+  auto &refY = inputWS->getSpectrum(0).y();
   std::vector<double> x_pos(refY.size());
```

Comment:
I think you can simply use `auto x_pos = inputWS->points(0);` here and get rid of the `if/else` block below.